doc/irmin-pack/index.mld

```diff
@@ -2,6 +2,8 @@
 
 The [irmin-pack] package provides a space-optimized, on-disk backend inspired by {{:https://git-scm.com/book/en/v2/Git-Internals-Packfiles} Git Packfiles}.
 
+[irmin-pack] is desigend and developed for a very specific use-case and requires backend specific integration from applications. Users interested in a more general and stable Irmin backend are recommended to use other backend such as {!Irmin_fs} or {!Irmin_git}.
```

Comment:
Let's remove the `and stable` part. :sweat_smile: As a back-end, it is quite stable (it has the heaviest production usage that I know of), but it is a bit more purpose-built in its intended usage and some of its API, as you point out.

Reply:
I meant stable in the sense of API stability. Of course as back-end it is stable :).

I've changed the note to following:

> Users interested in a more general Irmin backend with a more stable API are recommended to use other backend such as {!Irmin_fs} or {!Irmin_git}.

Is that ok?

I think the warning is fair. As a prospective Irmin user I would be very interested in API stability.

Reply:
Aha! I see the distinction you're making now.

I agree with giving guidance to users, but also don't want to unnecessarily guide them away from `irmin-pack` since I do think it is probably the most performant and stable backend given its extensive real-world usage.

When I think of the API of `irmin-pack` and `irmin`, here is how I break it down:

- Configuration. This is the most complex and application-specific for `irmin-pack`, but helped by the examples (thanks for adding another!).
- Core Irmin API. This is the same across all backends so not applicable to the guidance.
- The extensions to the store API. This area is arguably less stable (I'm not sure how it has changed historically) but this PR makes it much better from an external integration standpoint (we won't accidentally leak all our internal APIs now!).

If the above corresponds to what you are thinking when you think about the API and its stability, what do you think the best guidance to give is?

To me, it is something like:
> [irmin-pack] is designed and developed for a specific use-case and requires more in-depth backend configuration from applications. It offers additional functionality through extensions to the core Irmin API, but users interested in a more general Irmin backend with simpler configuration are recommended to use other backend such as {!Irmin_fs} or {!Irmin_git}.
>
> See [examples/irmin-pack] for examples of how to configure and use some of the additional functionality.

Reply:
> If the above corresponds to what you are thinking when you think about the API and its stability, what do you think the best guidance to give is?

I tend to agree with your break down. I just feel that a big API, even if it not used, is a considerable complexity factor that should be considered by prospective users.

In any case, I like your suggested wording and have pushed 25b40c0e34b7e733215823b29e161859a2fc6300 that uses it.

Thanks!